runner/full.go

```diff
@@ -254,3 +257,27 @@ func CollectAllServers(servers []*state.Server, globalCollectionOpts state.Colle
 
 	return
 }
+
+func pruneQueryIdentities(oldMap state.QueryIdentityMap) (newMap state.QueryIdentityMap) {
+	if len(oldMap) < 100000 {
+		return oldMap
+	}
+	slice := make([]state.QueryIdentity, 0, len(oldMap))
+	for _, identity := range oldMap {
+		slice = append(slice, identity)
+	}
+	sort.Slice(slice, func(i, j int) bool {
+		return slice[i].LastSeen.Before(slice[j].LastSeen)
+	})
+	for _, identity := range slice[:min(len(oldMap), 100000)] {
```

Comment:
will this `min(len(oldMap), 100000)` ever be `len(oldMap)` given that you have the early condition of `if len(oldMap) < 100000 {`?

Reply:
good catch, it looks like `len(oldMap)` will never be used